java/alizer-api/src/main/java/com/redhat/devtools/alizer/api/spi/framework/FrameworkDetectorProvider.java

```diff
@@ -10,14 +10,100 @@
+import com.redhat.devtools.alizer.api.utils.Utils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 
-public interface FrameworkDetectorProvider {
+public abstract class FrameworkDetectorProvider {
 
-    FrameworkDetectorProvider create();
+    public abstract FrameworkDetectorProvider create();
 
-    List<String> getSupportedLanguages();
+    public abstract List<String> getSupportedLanguages();
 
-    List<String> getFrameworks();
+    public abstract List<String> getFrameworks();
+
+    public boolean hasFramework(Language language) {
+        return getSupportedLanguages().stream().anyMatch(supported -> supported.equalsIgnoreCase(language.getName()))
+                && getFrameworks().stream().anyMatch(supported ->
+                language.getFrameworks().stream().anyMatch(langFramework -> langFramework.equalsIgnoreCase(supported)));
+    }
+
+    protected List<ServiceDescriptor> getServicesDescriptor(List<String> languagesAndFrameworks) throws IOException {
+        JsonNode node = Utils.getResourceAsJsonNode("/services.yml");
+
+        List<ServiceDescriptor> descriptors = new ArrayList<>();
+
+        for (Iterator<Map.Entry<String, JsonNode>> it = node.fields(); it.hasNext(); ) {
+            Map.Entry<String, JsonNode> entry = it.next();
+
+            String name = entry.getKey();
+            List<String> operators = new ArrayList<>();
+            Map<String, List<DependencyDescriptor>> dependencies = new HashMap<>();
+
+            JsonNode descriptorNode = entry.getValue();
+            if (descriptorNode.has("operators")) {
+                for(JsonNode operatorNode: descriptorNode.get("operators")) {
+                    operators.add(operatorNode.asText());
+                }
+            }
+            if (descriptorNode.has("dependencies")) {
+                for (Iterator<Map.Entry<String, JsonNode>> jt = descriptorNode.get("dependencies").fields(); jt.hasNext(); ) {
+                    Map.Entry<String, JsonNode> dependencyEntry = jt.next();
+                    String language = dependencyEntry.getKey();
+                    if (languagesAndFrameworks.stream().anyMatch(field -> field.equalsIgnoreCase(language))) {
+                        ArrayNode dependenciesNode = (ArrayNode) dependencyEntry.getValue();
+                        List<DependencyDescriptor> dependencyDescriptors  = extractDependenciesDescriptor(language, dependenciesNode);
+                        dependencies.put(language, dependencyDescriptors);
+                    }
+                }
+            }
+
+            descriptors.add(new ServiceDescriptor(name, operators, dependencies));
+        }
+        return descriptors;
+    }
+
+    private List<DependencyDescriptor> extractDependenciesDescriptor(String language, ArrayNode dependenciesNode) {
+        switch(language) {
+            case "java":
```

Comment:
Why those specific cases ? Language or framework specific stuff should be located in SPI, this will prevent others to contribute and limit alizer adoption